### PR TITLE
Feature/55184 update school audit trigger

### PR DESCRIPTION
**Pull request by jon-shipley**
* Update school audit trigger to handle multiple rows and deletes
* Update pupil audit trigger to handle deletes
* Update ms-sql version to latest
* Set ms-azure-edge as default db for local dev

### CODE REVIEW

File: db/migrations/schema-objects/20221122163818.do.fix-school-audit-trigger.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
+DROP TRIGGER IF EXISTS [mtc_admin].[schoolInsertUpdateAuditTrigger];
+
+go
+
+CREATE OR ALTER TRIGGER [schoolInsertUpdateAuditTrigger] ON [mtc_admin].[school]
```
Comment (GuyHarwood) [inline db/migrations/schema-objects/20221122163818.do.fix-school-audit-trigger.sql:5]: ive tried this locally and get the same error without the DROP first.  Bizarre.

maybe remove the OR ALTER and just stick to the traditional method? as the newer construct is clearly broken.

**Review by GuyHarwood — CHANGES_REQUESTED** [review]
all looks good, just the issues to address.

Also - should we remove the Foreign key from school audit to school on this PR?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   sqldb:
-    image: "mcr.microsoft.com/mssql/server:2019-CU16-GDR1-ubuntu-20.04"
+    image: "mcr.microsoft.com/mssql/server:2022-latest"
     #image: "mcr.microsoft.com/azure-sql-edge"
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply docker-compose.yml]: Done!

File: db/migrations/schema-objects/20221122163818.do.fix-school-audit-trigger.sql (hunk 1 above)
Comment (jon-shipley) [reply db/migrations/schema-objects/20221122163818.do.fix-school-audit-trigger.sql]: I've scoped the trigger name with the schema and this does the trick, as per https://github.com/jon-shipley/patterns/blob/master/mssql-idempotent-sql.md#Triggers

**Review by GuyHarwood — APPROVED** [review]
LGTM 🚀

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**